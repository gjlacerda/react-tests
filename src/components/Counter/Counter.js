import React, {Component} from 'react'

class Counter extends Component {
  state = {
    count: 0
  }

  increment = () => {
    this.setState(prevState => ({
      count: prevState.count + 1
    }))
  }

  decrement = () => {
    this.setState(prevState => ({
      count: prevState.count - 1
    }))
  }

  render() {
    return (
        <React.Fragment>
          <p>Current count: {this.state.count}</p>
          <button 
          className="increment"
          onClick={this.increment}>
            Increment count
          </button>
          <button
          className="decrement"
          onClick={this.decrement}>
            Decrement count
          </button>
        </React.Fragment>
    )
  }
}

export default Counter;