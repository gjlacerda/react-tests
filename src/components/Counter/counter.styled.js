import styled from 'styled-components'

const Paragraph = styled.p`
  color: #333;
`
Paragraph.displayName = 'Paragraph'

export {
  Paragraph
}