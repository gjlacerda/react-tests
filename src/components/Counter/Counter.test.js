import React from 'react'
import Counter from './Counter'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

describe('Counter component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Counter/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('starts with a count of 0', () => {
    const wrapper = shallow(<Counter/>)
    const text = wrapper.find('Paragraph').text()
    expect(text).toEqual('Current count: 0')
  })

  it('increments the count when the button is clicked', () => {
    const wrapper = shallow(<Counter/>)
    const button = wrapper.find('button.increment')
    button.simulate('click')
    const text = wrapper.find('Paragraph').text()
    expect(text).toEqual('Current count: 1')
  })

  it('decrements the count when the decrement button is clicked', () => {
    const wrapper = shallow(<Counter/>)
    const decrementButton = wrapper.find('button.decrement')
    decrementButton.simulate('click')
    const text = wrapper.find('Paragraph').text()
    expect(text).toEqual('Current count: -1')
  })
})