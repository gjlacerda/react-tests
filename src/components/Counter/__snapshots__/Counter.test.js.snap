// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches the snapshot 1`] = `
Array [
  <p
    className="sc-bdVaJa dsMjwO"
  >
    Current count: 
    0
  </p>,
  <button
    className="increment"
    onClick={[Function]}
  >
    +++
  </button>,
  <button
    className="decrement"
    onClick={[Function]}
  >
    ---
  </button>,
]
`;
